contentcuration/contentcuration/viewsets/contentnode.py

```diff
@@ -0,0 +1,208 @@
+
+
+class ContentNodeFilter(FilterSet):
+    ids = CharFilter(method="filter_ids")
+    channel_root = CharFilter(method="filter_channel_root")
+
+    class Meta:
+        model = ContentNode
+        fields = ("parent", "ids", "kind", "channel_root")
+
+    def filter_ids(self, queryset, name, value):
+        try:
+            # Limit SQL params to 50 - shouldn't be fetching this many
+            # ids at once
+            return queryset.filter(pk__in=value.split(",")[:50])
+        except ValueError:
+            # Catch in case of a poorly formed UUID
+            return queryset.none()
+
+    def filter_channel_root(self, queryset, name, value):
+        return queryset.filter(
+            parent=Channel.objects.filter(pk=value).values_list(
+                "main_tree__id", flat=True
+            )
+        )
+
+
+class SQCount(Subquery):
+    # Include ALIAS at the end to support Postgres
+    template = "(SELECT COUNT(%(field)s) FROM (%(subquery)s) AS %(field)s__sum)"
+    output_field = IntegerField()
+
+
+class ContentNodeSerializer(ModelSerializer):
+    """
+    This is a write only serializer - we leverage it to do create and update
+    operations, but read operations are handled by the Viewset.
+    """
+
+    class Meta:
+        model = ContentNode
+        fields = ("id", "title", "kind", "content_id", "language", "role_visibility")
+        read_only_fields = ("id",)
+
+
+def retrieve_thumbail_src(item):
+    """ Get either the encoding or the url to use as the <img> src attribute """
+    try:
+        if item.get("thumbnail_encoding"):
+            return json.loads(item.get("thumbnail_encoding")).get("base64")
+    except ValueError:
+        pass
+    if (
+        item["thumbnail_checksum"] is not None
+        and item["thumbnail_extension"] is not None
+    ):
+        return generate_storage_url(
+            "{}.{}".format(item["thumbnail_checksum"], item["thumbnail_extension"])
+        )
+    return None
+
+
+class SummaryContentNodeViewSet(ValuesViewset):
+    queryset = ContentNode.objects.all()
+    serializer_class = ContentNodeSerializer
+    filter_backends = (DjangoFilterBackend,)
+    filter_class = ContentNodeFilter
+    values = (
+        "id",
+        "title",
+        "kind",
+        "content_id",
+        "language_id",
+        "role_visibility",
+        "total_count",
+        "resource_count",
+        "coach_count",
+        "thumbnail_checksum",
+        "thumbnail_extension",
+        "thumbnail_encoding",
+        "published",
+        "modified",
+        "parent",
+        "has_children",
+        # "valid",
+    )
+
+    field_map = {"language": "language_id", "thumbnail_src": retrieve_thumbail_src}
+
+    def get_queryset(self):
+        queryset = ContentNode.objects.all()
+        return self.prefetch_queryset(queryset)
+
+    def annotate_queryset(self, queryset):
+        queryset = queryset.annotate(total_count=(F("rght") - F("lft")) / 2)
```

Comment:
In `MPTTModel.get_descendant_count()` they subtract 1 from the right and left, so it would be `(F("rght") - F("lft") - 1) / 2`. Any reason for the difference?